cypress/integration/code-schools.spec.js

```diff
@@ -14,4 +14,31 @@ describe('code schools', function() {
     // 40 is arbitrary, but it proves that the API is working and leading to rendered content
     cy.get('[data-testid="SchoolCard"]').should('have.length.greaterThan', 40);
   });
+
+  it('should render Cincy and Tech Elevator after selecting Ohio', () => {
+    cy.wait('@codeSchools');
+    cy.get('input#react-select-state_select-input')
+      .type('Ohio', { force: true })
+      .type('{enter}');
+
+    cy.get('[data-testid="SchoolCard"]').should('have.length', 2);
+  });
+
+  it('should render GI Bill accepted schools after hitting VA button', () => {
+    cy.wait('@codeSchools');
+    cy.contains('VA Approved Schools').click();
+
+    cy.get('[data-testid="SchoolCard"]').should('have.length.greaterThan', 20);
```

Comment:
This won't quite be what I'm looking for.

Does your assertion actually guarantee that the >20 SchoolCards you've rendered take the GI Bill? What if, for example, the button broke... You'd just be matching the first test, and that is what's called a "false positive". A test that passes, but shouldn't.

If you look at the source code for the SchoolCard component, you can search for references to the GI Bill to see what sets it apart from other SchoolCards. You could assert that the text `GI Bill` exists, but that would be true for all cards (as it acts as a label here:

```javascript
<div className={styles.cardBlock}>
  <span className={styles.cardBlockTitle}>Accepts GI Bill</span>
  {hasGiBill ? 'Yes' : 'No'}
</div>
```

The ribbon, however, is conditionally rendered based upon whether or not it takes the GI Bill. I recommend passing `data-testid` to the div that wraps the ribbon, and find assert that _every_ rendered SchoolCard has it.

Reply:
Completed with data-testid in GI Bill Ribbon